src/main/java/gov/cdc/sdp/cbr/queue/DatabaseQueueConsumer.java

```diff
@@ -66,14 +66,14 @@ public Integer doInPreparedStatement(PreparedStatement ps) throws SQLException {
     }
 
     public DatabaseQueueConsumer(Endpoint endpoint, DataSource ds, Processor processor, String tn, int delay,
-            int initialDelay) {
+            int initialDelay, int limit, int maxAttempts) {
         super(endpoint, processor);
         this.jdbcTemplate = new JdbcTemplate(ds);
         this.tableName = tn;
 
-        this.query = "with batch as (" + "SELECT * " + "FROM  " + tableName + " WHERE  status = 'queued' "
-                + "LIMIT  100 " + "FOR    UPDATE SKIP LOCKED" + "   ) " + "UPDATE " + tableName + " t "
-                + "SET    status = 'sending' " + "FROM   batch " + "WHERE  t.id = batch.id " + "RETURNING *;";
+        this.query = "with batch as (SELECT * FROM " + tableName + " WHERE status = 'queued' AND attempts<="
```

Comment:
If the attempts fail, the status will no longer be queued.  Can you update either the onConsumeFailed query to only set status to failed if attempts is reached or update this query to include failed status?  Note that since the select sets the status to sending, onConsumeFailed will need to revert to "queued" if attempts are not reached in the first case.